File: result/metagpt/claude/humaneval/0.c
#include <stdio.h>
#include <math.h>
#include <stdbool.h>

bool has_close_elements(float* numbers, int size, float threshold) {
    int i, j;
    
    for (i = 0; i < size; i++) {
        for (j = i + 1; j < size; j++) {
            if (fabs(numbers[i] - numbers[j]) < threshold) {
                return true;
            }
        }
    }
    
    return false;
}

File: result/metagpt/deepseek/humaneval/0.rs
pub fn has_close_elements(numbers: &[f32], threshold: f32) -> bool {
    numbers.iter().enumerate().any(|(i, &x)| {
        numbers[i+1..].iter().any(|&y| (x - y).abs() < threshold)
    })
}

#[cfg(test)]
mod tests {
    use super::*;
    
    #[test]
    fn test_has_close_elements() {
        assert!(has_close_elements(&[1.0, 2.0, 3.0, 4.0, 1.5], 0.5));
        assert!(!has_close_elements(&[1.0, 2.0, 3.0, 4.0, 5.0], 0.5));
    }
}